x-pack/legacy/plugins/infra/public/pages/infrastructure/metrics_explorer/use_metric_explorer_state.ts

```diff
@@ -100,6 +112,18 @@ export const useMetricsExplorerState = (
     [options]
   );
 
+  const onViewStateChange = useCallback((vs: MetricExplorerViewState) => {
+    if (vs.chartOptions) {
+      setChartOptions(vs.chartOptions);
+    }
+    if (vs.currentTimerange) {
+      setTimeRange(vs.currentTimerange);
+    }
+    if (vs.options) {
+      setOptions(vs.options);
+    }
+  }, []);
```

Comment:
Add `setOptions`, `setChartOptions`, and `setTimerange` to list of dependencies.